File: Easy/maximum-average-subarray-i.py
class Solution:     # 1521ms
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        prev = 0
        for i in range(k):
            prev = ((prev * i) + nums[i]) / (i + 1)
        ret = prev
        for i in range(len(nums) - k):
            prev = ((prev * k) - nums[i] + nums[i + k]) / k
            ret = max(prev, ret)
        return ret

File: Easy/maximum-average-subarray-i.java
class Solution {        // 4ms
    public double findMaxAverage(int[] nums, int k) {
        double prev = 0;
        for(int i = 0; i < k; i++){
            prev += nums[i];
        }
        double ret = prev;
        for(int i = 0; i < nums.length - k; i++){
            prev += nums[i + k] - nums[i];
            ret = Math.max(prev, ret);
        }
        return ret / k;
    }
}